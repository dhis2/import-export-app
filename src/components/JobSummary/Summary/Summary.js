import i18n from '@dhis2/d2-i18n'
import { NoticeBox } from '@dhis2/ui'
import PropTypes from 'prop-types'
import React from 'react'
import { typeReportParse } from '../helper.js'
import { SingleSummary } from '../SingleSummary/SingleSummary.js'
import { TypeReportSummary } from '../TypeReportSummary/TypeReportSummary.js'
import styles from './Summary.module.css'

const extractStats = (summary) => {
    if (summary.responseType === 'ImportSummaries') {
        const { imported, deleted, ignored, updated, total } = summary
        return { imported, deleted, ignored, updated, total }
    } else if (summary.importCount) {
        const { imported, deleted, ignored, updated } = summary.importCount
        const total = imported + deleted + ignored + updated
        return { imported, deleted, ignored, updated, total }
    } else if (summary.stats) {
        return summary.stats
    }
}

const Summary = ({ summary, importType, isDryRun }) => {
    if (isDryRun && importType === 'TRACKER_IMPORT_JOB') {
        const { ignored, total } = extractStats(summary)
        return (
            <div data-test="job-summary-summary" className={styles.container}>
                <div
                    className={styles.rest}
                    data-test="job-summary-summary-rest"
                >
                    <NoticeBox title="Summary">
                        {i18n.t(
                            `${ignored} entities will be ignored out of ${total} entities‚Å†`
                        )}
                    </NoticeBox>
                </div>
            </div>
        )
    }

    if (summary.typeReports) {
        const overviewStats = {
            ...summary.stats,
            imported: summary.stats.created,
        }
        const { stats, messages } = typeReportParse(summary.typeReports)
        return (
            <div
                data-test="job-summary-summary"
                className={styles.typeReportSummary}
            >
                <TypeReportSummary
                    overviewStats={overviewStats}
                    stats={stats}
                    messages={messages}
                />
            </div>
        )
    }

    const importCount = extractStats(summary)

    const overviewSummary = (
        <SingleSummary
            importType={importType}
            importCount={importCount}
            status={summary.status}
            description={summary.description}
            validationReport={summary.validationReport}
            conflicts={
                summary.conflicts &&
                (summary.conflicts.length || null) &&
                summary.conflicts
            }
        />
    )

    const allSummaries =
        summary.responseType === 'ImportSummaries' && summary.importSummaries
            ? summary.importSummaries.map((s, i) => {
                  const importCount = extractStats(s)
                  return (
                      <SingleSummary
                          key={`single-summary-${i}`}
                          importType={importType}
                          importCount={importCount}
                          status={s.status}
                          description={s.description}
                          conflicts={s.conflicts}
                          id={`${i + 1}`}
                      />
                  )
              })
            : null

    return (
        <div data-test="job-summary-summary" className={styles.container}>
            {overviewSummary}
            <div className={styles.rest} data-test="job-summary-summary-rest">
                {allSummaries}
            </div>
        </div>
    )
}

Summary.propTypes = {
    summary: PropTypes.object.isRequired,
    importType: PropTypes.string,
    isDryRun: PropTypes.bool,
}

export { Summary }
