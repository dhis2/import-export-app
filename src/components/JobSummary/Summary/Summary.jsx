import PropTypes from 'prop-types'
import React from 'react'
import { typeReportParse } from '../helper.js'
import { SingleSummary } from '../SingleSummary/SingleSummary.jsx'
import { TypeReportSummary } from '../TypeReportSummary/TypeReportSummary.jsx'
import styles from './Summary.module.css'

const extractStats = (summary) => {
    if (summary.responseType == 'ImportSummaries') {
        const { imported, deleted, ignored, updated, total } = summary
        return { imported, deleted, ignored, updated, total }
    } else if (summary.importCount) {
        const { imported, deleted, ignored, updated } = summary.importCount
        const total = imported + deleted + ignored + updated
        return { imported, deleted, ignored, updated, total }
    } else if (summary.stats) {
        return summary.stats
    }
}

const Summary = ({ summary, importType }) => {
    // gml import type object return
    if (summary.typeReports) {
        const overviewStats = {
            ...summary.stats,
            imported: summary.stats.created,
        }
        const { stats, messages } = typeReportParse(summary.typeReports)
        return (
            <div
                data-test="job-summary-summary"
                className={styles.typeReportSummary}
            >
                <TypeReportSummary
                    overviewStats={overviewStats}
                    stats={stats}
                    messages={messages}
                />
            </div>
        )
    }

    const importCount = extractStats(summary)

    const overviewSummary = (
        <SingleSummary
            importType={importType}
            importCount={importCount}
            status={summary.status}
            description={summary.description}
            validationReport={summary.validationReport}
            conflicts={
                summary.conflicts &&
                (summary.conflicts.length || null) &&
                summary.conflicts
            }
        />
    )
    const allSummaries =
        summary.responseType == 'ImportSummaries' && summary.importSummaries
            ? summary.importSummaries.map((s, i) => {
                  const importCount = extractStats(s)
                  return (
                      <SingleSummary
                          key={`single-summary-${i}`}
                          importType={importType}
                          importCount={importCount}
                          status={s.status}
                          description={s.description}
                          conflicts={s.conflicts}
                          id={`${i + 1}`}
                      />
                  )
              })
            : null

    return (
        <div data-test="job-summary-summary" className={styles.container}>
            {overviewSummary}
            <div className={styles.rest} data-test="job-summary-summary-rest">
                {allSummaries}
            </div>
        </div>
    )
}

Summary.propTypes = {
    summary: PropTypes.object.isRequired,
    importType: PropTypes.string,
}

export { Summary }
