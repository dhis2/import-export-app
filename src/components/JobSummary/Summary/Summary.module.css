.container {
    max-height: 440px;
    overflow: auto;
    margin-bottom: 15px;
}

.rest,
.typeReportSummary {
    max-height: 440px;
    overflow: auto;
    margin-bottom: 15px;
}
