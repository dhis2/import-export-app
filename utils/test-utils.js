import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
