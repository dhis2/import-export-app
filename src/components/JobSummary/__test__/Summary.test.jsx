import React from 'react'
import { render } from 'test-utils'

import { allJobs } from '../../JobOverview/__test__/data.js'
import { Summary } from '../Summary/Summary.jsx'

describe('different job type summaries', () => {
    const oneOfEach = [
        allJobs.find((j) => j.importType == 'DATAVALUE_IMPORT'),
        allJobs.find((j) => j.importType == 'EVENT_IMPORT'),
        allJobs.find((j) => j.importType == 'GML_IMPORT'),
        allJobs.find((j) => j.importType == 'METADATA_IMPORT'),
    ]

    oneOfEach.forEach((j) =>
        it(`matches snapshot - ${j.importType}`, () => {
            const { asFragment } = render(<Summary summary={j.summary} />)
            expect(asFragment()).toMatchSnapshot()
        })
    )
})
