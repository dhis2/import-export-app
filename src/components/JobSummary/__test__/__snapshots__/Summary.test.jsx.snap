// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`different job type summaries matches snapshot - DATAVALUE_IMPORT 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-test="job-summary-summary"
  >
    <div>
      <div
        class="container"
        data-test="job-summary-single-summary"
      >
        <span
          class="label"
        >
          Summary
        </span>
        <table
          class="jsx-2430604489 "
          data-test="dhis2-uicore-table"
        >
          <thead
            data-test="dhis2-uicore-tablehead"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerowhead"
            >
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Status
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Description
              </th>
            </tr>
          </thead>
          <tbody
            data-test="dhis2-uicore-tablebody"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                SUCCESS
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                Import process completed successfully
              </td>
            </tr>
          </tbody>
        </table>
        <table
          class="jsx-2430604489 "
          data-test="dhis2-uicore-table"
        >
          <thead
            data-test="dhis2-uicore-tablehead"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerowhead"
            >
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Created
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Deleted
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Ignored
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Updated
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Total
              </th>
            </tr>
          </thead>
          <tbody
            data-test="dhis2-uicore-tablebody"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="rest"
      data-test="job-summary-summary-rest"
    />
  </div>
</DocumentFragment>
`;

exports[`different job type summaries matches snapshot - EVENT_IMPORT 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-test="job-summary-summary"
  >
    <div>
      <div
        class="container"
        data-test="job-summary-single-summary"
      >
        <span
          class="label"
        >
          Summary
        </span>
        <table
          class="jsx-2430604489 "
          data-test="dhis2-uicore-table"
        >
          <thead
            data-test="dhis2-uicore-tablehead"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerowhead"
            >
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Created
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Deleted
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Ignored
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Updated
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Total
              </th>
            </tr>
          </thead>
          <tbody
            data-test="dhis2-uicore-tablebody"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                1
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                1
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="rest"
      data-test="job-summary-summary-rest"
    >
      <div>
        <div
          class="container"
          data-test="job-summary-single-summary"
        >
          <span
            class="label"
          >
            Summary #1
          </span>
          <table
            class="jsx-2430604489 "
            data-test="dhis2-uicore-table"
          >
            <thead
              data-test="dhis2-uicore-tablehead"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerowhead"
              >
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Status
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Description
                </th>
              </tr>
            </thead>
            <tbody
              data-test="dhis2-uicore-tablebody"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerow"
              >
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  ERROR
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  Event date is required. 
                </td>
              </tr>
            </tbody>
          </table>
          <table
            class="jsx-2430604489 "
            data-test="dhis2-uicore-table"
          >
            <thead
              data-test="dhis2-uicore-tablehead"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerowhead"
              >
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Created
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Deleted
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Ignored
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Updated
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Total
                </th>
              </tr>
            </thead>
            <tbody
              data-test="dhis2-uicore-tablebody"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerow"
              >
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  1
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  1
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`different job type summaries matches snapshot - GML_IMPORT 1`] = `
<DocumentFragment>
  <div
    class="typeReportSummary"
    data-test="job-summary-summary"
  >
    <div>
      <div>
        <div
          class="container"
          data-test="job-summary-single-summary"
        >
          <span
            class="label"
          >
            Summary
          </span>
          <table
            class="jsx-2430604489 "
            data-test="dhis2-uicore-table"
          >
            <thead
              data-test="dhis2-uicore-tablehead"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerowhead"
              >
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Created
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Deleted
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Ignored
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Updated
                </th>
                <th
                  class="jsx-1796049005 "
                  data-test="dhis2-uicore-tablecellhead"
                >
                  Total
                </th>
              </tr>
            </thead>
            <tbody
              data-test="dhis2-uicore-tablebody"
            >
              <tr
                class="jsx-3124144591 zebraStriping"
                data-test="dhis2-uicore-tablerow"
              >
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
                <td
                  class="jsx-996330601 "
                  data-test="dhis2-uicore-tablecell"
                >
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="container"
        data-test="job-summary-type-count"
      >
        <span
          class="label"
        >
          Details by type
        </span>
        <table
          class="jsx-2430604489 "
          data-test="dhis2-uicore-table"
        >
          <thead
            data-test="dhis2-uicore-tablehead"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerowhead"
            >
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Type
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Created
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Deleted
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Ignored
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Updated
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Total
              </th>
            </tr>
          </thead>
          <tbody
            data-test="dhis2-uicore-tablebody"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                DefaultGmlImportService
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="container"
        data-test="job-summary-messages"
      >
        <span
          class="label"
        >
          Messages
        </span>
        <table
          class="jsx-2430604489 "
          data-test="dhis2-uicore-table"
        >
          <thead
            data-test="dhis2-uicore-tablehead"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerowhead"
            >
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                UID
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Type
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Property
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Message
              </th>
            </tr>
          </thead>
          <tbody
            data-test="dhis2-uicore-tablebody"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              />
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                DefaultGmlImportService
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              />
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                Failed to validate job runtime - \`GML import failed: Content is not allowed in prolog. On line 1 column 1.\`
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`different job type summaries matches snapshot - METADATA_IMPORT 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-test="job-summary-summary"
  >
    <div>
      <div
        class="container"
        data-test="job-summary-single-summary"
      >
        <span
          class="label"
        >
          Summary
        </span>
        <table
          class="jsx-2430604489 "
          data-test="dhis2-uicore-table"
        >
          <thead
            data-test="dhis2-uicore-tablehead"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerowhead"
            >
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Created
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Deleted
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Ignored
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Updated
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Total
              </th>
            </tr>
          </thead>
          <tbody
            data-test="dhis2-uicore-tablebody"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                0
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="rest"
      data-test="job-summary-summary-rest"
    />
  </div>
</DocumentFragment>
`;
